### DISCUSSION TITLE
Maui packages not in sync with the latest stable visual studio release

### DISCUSSION THREAD

**Discussion by thombrink** [CommunityToolkit/Maui#1850] [Q&A] [answered]
I'm getting downgrade warnings because the toolkit nuget uses the 8.0.14, but VS 17.9 is still using 8.0.7, Microsoft recommends to use  `$(MauiVersion)` and I don't want to use the preview version of VS.

I would expect that if 8.0.14 is used, the toolkit nuget would be a preview version?

Thanks in advance.

**Comment by bijington** [comment]
17.9.3 ships with MAUI 8.0.7 SR2. The latest preview ships with MAUI 8.0.21 SR4. I don't know what exact version is SR3 but I can't find anything in the Microsoft release notes for it.

That being said you can update the version of .NET MAUI without updating Visual Studio.

Something like

dotnet workload install maui --sdk-version=8.0.14

**Reply by thombrink** [reply]
Yeah thats what I meant. But why release a nuget that's not yet supported by a stable VS release, without having to execute commands manually? I don't feel like having to explain this to my co-workers. One of them already replaced the `$(MauiVersion)` with the 8.0.14 and therefore undermined the point of having a `$(MauiVersion)`, because he already installed all VS updates and he still got warnings (which in our case are errors) and caused the build to fail.

So with a bit of a background, this is in a nutshell why I wondered why those nugets aren't also preview nuget's.

Thanks for the workaround, for now we will use that, but if you would like to consider labeling such nugets as a preview version, that would be great!

**Reply by bijington** [reply, marked as answer]
I don't think we should mark our releases as preview because we are not using any preview releases of .NET MAUI - 8.0.14 is a stable release and there are no preview features of the toolkit.

That being said I understand the frustration of not having the tooling up to date with our dependencies. There are no easy answers here as developers won't want to run command line tools to solve the issue and then we as maintainers don't want to be held up releasing a bug fix based on a piece of toolings release cycle. Then if you imagine there is Visual Studio, Visual Studio Code and Rider to name a few IDEs it could become a lot trickier.

We have our stand up tomorrow and I will add the discussion label so we talk about it to see what the team think.

**Reply by thombrink** [reply]
Ah I understand! Thanks for pointing out why currently works like this. Releasing a preview version would indeed not fix it.

Thanks for taking the time to explain this and I'm looking forward to the conclusion of the standup!

**Comment by TheCodeTraveler** [comment]
This is good to know! Thanks for raising it, @thombrink.

I do all my dev on a Mac using Rider, and I always [update the .NET SDK manually](https://dotnet.microsoft.com/en-us/download/dotnet/8.0) and subsequently update MAUI manually, `sudo dotnet workload install maui`. 

I didn't realize that Visual Studio lagged behind in auto-updating MAUI for devs. 

Agreed, @bijington, we should chat about this on our standup to figure out the best plan for our `Microsoft.Maui.Controls` NuGet dependency going forward 💯

**Comment by mattleibow** [comment]
I opened this issue, please vote on it and share your pains so they can prioritize it: https://developercommunity.visualstudio.com/t/Installing-a-NuGet-that-depends-on-later/10651482

Also, add comments and ideas so we can bootstrap the thinking and potential fixes faster.

**Comment by CliffAgius** [comment]
Closed as Answered.
